### PR TITLE
Allow specifying cloudinit_cdrom_storage to add a CloudInit Drive

### PR DESCRIPTION
**Pull request by ianlevesque**
If you have a Proxmox QEMU template that supports cloud-init (perhaps you ran 'apt-get install cloud-init' during installation) then you need a "CloudInit Drive" attached to the VM to be able to pass the cloud-init parameters to the guest.  This adds a parameter 'cloudinit_cdrom_storage' that will configure a CloudInit Drive stored on the storage you specify. For example: cloudinit_cdrom_storage = "local-lvm".

Proxmox has an unfortunate "magic syntax" for these devices. Their documentation suggests that you do a 'config set' on a created template for something like 'local-lvm:cloudinit' to make it work.  Using a disk {} clause in the existing proxmox resource hierarchy doesn't work.

### CODE REVIEW

**Comment by ianlevesque** [comment]
I think this CI is just broken. If anyone knows what is wrong with this PR let me know please.

**Comment by pryorda** [comment]
@ianlevesque Are you running into this issue?

```
create full clone of drive ide2 (local-lvm:vm-9000-cloudinit)
TASK ERROR: clone failed: error with cfs lock 'storage-...: unable to parse volume filename 'vm-100-cloudinit' 
```

**Comment by ianlevesque** [comment]
I think the latest 6.1 changed something but I haven’t had time to investigate.

**Comment by msbence** [comment]
No, this error is present in 6.3-3 as well, but they will fix it.
See: [https://lists.proxmox.com/pipermail/pve-devel/2020-December/046315.html](https://lists.proxmox.com/pipermail/pve-devel/2020-December/046315.html)

tl;dr: Cloning of cloudinit images will fail on `raw` type, works only for `qcow2`.

**Comment by msbence** [comment]
The issue is now fixed with the latest qemu-server. @ianlevesque can you re-trigger the CI?

```
qemu-server (6.3-3) pve; urgency=medium
  [...]
  * cloudinit: fix cloning/restoring of cloudinit disks in raw format
 -- Proxmox Support Team <support@proxmox.com>  Tue, 15 Dec 2020 16:33:01 +0100
```

**Comment by ianlevesque** [comment]
TBH I haven't used this in a year, but I rebased and re-pushed if it helps

**Comment by msbence** [comment]
It helped! Thank you :)
